src/OpenTelemetry/CompositeProcessor.cs

```diff
@@ -69,6 +76,21 @@ public override void OnEnd(T data)
         }
     }
 
+#if EXPOSE_EXPERIMENTAL_FEATURES
+    /// <inheritdoc/>
+    [Experimental(DiagnosticDefinitions.ExtendedBaseProcessorExperimentalApi, UrlFormat = DiagnosticDefinitions.ExperimentalApiUrlFormat)]
+    public override void OnEnding(T data)
+    {
+        for (var cur = this.Head; cur != null; cur = cur.Next)
+        {
+            if (typeof(ExtendedBaseProcessor<T>).IsAssignableFrom(cur.Value.GetType()))
+            {
+                ((ExtendedBaseProcessor<T>)cur.Value).OnEnding(data);
+            }
+        }
```

Comment:
Would this work to avoid reflection and casting?

```suggestion
        for (var current = this.Head; current != null; cur = current.Next)
        {
            if (current.Value is ExtendedBaseProcessor<T> processor)
            {
                processor.OnEnding(data);
            }
        }
```